lib/builder/build.go

```diff
@@ -121,7 +121,8 @@ func checkBuildEnv() error {
 		return trace.BadParameter("tele build is not supported on %v, only "+
 			"Linux is supported", runtime.GOOS)
 	}
-	client, err := docker.NewClient(constants.DockerEngineURL)
+	endpoint := utils.GetenvWithDefault("DOCKER_HOST", constants.DockerEngineURL)
+	client, err := docker.NewClient(endpoint)
```

Comment:
I think that Vendorer (inside Builder) also uses the hardcoded default Docker URL (https://github.com/gravitational/gravity/blob/master/lib/builder/builder.go#L322). I would consider passing Docker URL to the Builder via `builder.Config`.